GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -359,6 +360,50 @@ void ManageWorktreeSupport()
             }
         }
 
+        private void FillNextPullActionAsDefaultToolStripMenuItems()
+        {
+            var setDefaultPullActionDropDown = (ToolStripDropDownMenu)setDefaultPullButtonActionToolStripMenuItem.DropDown;
+
+            // Show both Check and Image margins in a menu
+            setDefaultPullActionDropDown.ShowImageMargin = true;
+            setDefaultPullActionDropDown.ShowCheckMargin = true;
+
+            // Prevent submenu from closing while options are changed
+            setDefaultPullActionDropDown.Closing += (sender, args) =>
+            {
+                if (args.CloseReason == ToolStripDropDownCloseReason.ItemClicked)
+                {
+                    args.Cancel = true;
+                }
+            };
+
+            foreach (object pullActionMenuEntry in toolStripButtonPull.DropDownItems)
+            {
+                if (pullActionMenuEntry is ToolStripMenuItem pullActionMenuItem && Enum.TryParse(pullActionMenuItem.Tag as string, out AppSettings.PullAction pullAction))
```

Comment:
We can probably combine `if` with `foreach`:

```
foreach (var menuItem in toolStripButtonPull.DropDownItems
                                            .OfType<ToolStripMenuItem>()
                                            .Where(tsmi => tsmi.Tag is AppSettings.PullAction)) 
{
}
```

This should make it simpler and no need to `Enum.Parse`.

(I'm writing this in a browser, not 100% it is correct)

Reply:
Hmm, not sure about being able to get rid of Enum.Parse. The thing is I wanted to set the Tags in designer and that only supports strings (or I don't know the correct syntax). I didn't want to litter FormBrowse.cs more than necessary. So Tags are strings and need parsing.

Reply:
Setting strings in the designer file that have to match member names in an enum seems suboptimal in the face of future discovery/refactoring. We have code elsewhere that sets tags (etc) in the constructor after `InitializeComponent`. That way you could also set the enum value directly, rather than a string, and avoid the parse.